### PR TITLE
Fix the rename-to-exe logic for the MinGW build

### PR DESCRIPTION
**Pull request by kholia**
This renaming is breaking quite a few things.

This PR tries to fix https://github.com/magnumripper/JohnTheRipper/issues/2397 to some extent.

### CODE REVIEW

**Comment by kholia** [comment]
@claudioandre What do you think about this PR? Does it solve more problems than it creates?

**Comment by kholia** [comment]
Once we clear up the renaming problem(s), it seems that the MinGW builds are ready for public consumption.

**Comment by kholia** [comment]
I am tweaking this stuff now. Please DO NOT MERGE.

**Comment by solardiz** [comment]
On Fri, Jan 27, 2017 at 01:02:15AM -0800, magnum wrote:
> -for f in `ls -l | grep wxr | grep -v [\.][epr] | cut -c 46-` ; do mv $f $f.exe ; done
> -for f in benchmark-unify mailer makechr relbench ; do mv $f.exe $f ; done
> +# for f in `ls -l | grep wxr | grep -v [\.][epr] | cut -c 46-` ; do mv $f $f.exe ; done
> +# for f in benchmark-unify mailer makechr relbench ; do mv $f.exe $f ; done
>  for f in *.exe ; do x86_64-w64-mingw32-strip $f ; done
> 
> Wow, that `cut -c 46-` doesn't look very safe to me. Or does POSIX specify exact column widths?

Can we use "test -x" instead?  Like:

for f in `echo *.[^epr]`; do test -x $f && mv $f $f.exe; done

(totally untested)

**Comment by kholia** [comment]
Since only a handful of files are missing the .exe extension, I prefer the simple and explicit `for f in genmkvpwd mkvcalcproba calc_stat tgtsnarf raw2dyna uaf2john wpapcap2john cprepair ; do mv $f $f.exe ; done` construct.

File: src/CircleCI-MinGW.sh [hunk 1]
```diff
@@ -38,8 +38,8 @@ mv ../run/john ../run/john.exe
 
 # the mingw build does not name many exe files correctly.  Fix that.  Also strip all exe files for distro.
 cd ../run
-for f in `ls -l | grep wxr | grep -v [\.][epr] | cut -c 46-` ; do mv $f $f.exe ; done
-for f in benchmark-unify mailer makechr relbench ; do mv $f.exe $f ; done
+# for f in `ls -l | grep wxr | grep -v [\.][epr] | cut -c 46-` ; do mv $f $f.exe ; done
+# for f in benchmark-unify mailer makechr relbench ; do mv $f.exe $f ; done
 for f in *.exe ; do x86_64-w64-mingw32-strip $f ; done
```
Comment (kholia) [inline src/CircleCI-MinGW.sh]: This for loop was generating a lot of error messages on my local Fedora 24 box. This whole thing is indeed very unsafe.

**Comment by kholia** [comment]
Updated commit message is,

```
Fix rename-to-exe logic for the MinGW build
    
This was breaking quite a few things.
    
This commit also fixes the 32-bit legacy build of vncpcap2john by
propagating the "-m32" flag correctly.
```

**Pull request merged**